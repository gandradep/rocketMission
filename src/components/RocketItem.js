import React from 'react';
import classes from './Rocket.module.css';
import ReserveBtn from './ReserveBtn';
import RocketBadge from './RocketBadge';

const RocketItem = (props) => {
  // eslint-disable-next-line react/prop-types
  const { image, name, description, id, reserved } = props;

  return (
    <div className={`${classes.items} dFlex`}>
      <div>
        <img src={image} alt="" className={classes.rocket_img} />
      </div>
      <div className={classes.rocket_description}>
        <h3 className={classes.rocket_name}>{name}</h3>
        {reserved ? <RocketBadge /> : ''}
        <p>{description}</p>
        {!reserved && (
          <ReserveBtn id={id} reserved={reserved}>
            Reserve Rocket
          </ReserveBtn>
        )}
        {reserved && (
          <ReserveBtn id={id} reserved={reserved}>
            Cancel Reservation
          </ReserveBtn>
        )}
      </div>
    </div>
  );
};

export default RocketItem;
