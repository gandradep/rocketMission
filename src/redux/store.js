import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from './missions/missions';
import rocketReducer from './rocketAction';

const store = configureStore({
  reducer: {
    rocket: rocketReducer,
    missions: missionsReducer,
  },
});

export default store;
