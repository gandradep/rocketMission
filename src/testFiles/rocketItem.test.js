import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import RocketItem from '../components/RocketItem';

describe('Appcomponents snapshots', () => {
  it('App renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <RocketItem />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
