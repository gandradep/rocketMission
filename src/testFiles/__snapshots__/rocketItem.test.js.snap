// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appcomponents snapshots App renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="items dFlex"
      >
        <div>
          <img
            alt=""
            class="rocket_img"
          />
        </div>
        <div
          class="rocket_description"
        >
          <h3
            class="rocket_name"
          />
          <p />
          <button
            class="btn"
            type="button"
          >
            Reserve Rocket
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="items dFlex"
    >
      <div>
        <img
          alt=""
          class="rocket_img"
        />
      </div>
      <div
        class="rocket_description"
      >
        <h3
          class="rocket_name"
        />
        <p />
        <button
          class="btn"
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
